src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -190,23 +195,37 @@ export async function filterLogEventsFromUriComponents(
     logGroupInfo: CloudWatchLogsGroupInfo,
     parameters: CloudWatchLogsParameters,
     nextToken?: string
-): Promise<CloudWatchLogsResponse> {
+): Promise<CloudWatchLogsResponse | void> {
     const client = new DefaultCloudWatchLogsClient(logGroupInfo.regionName)
 
-    const response = await client.filterLogEvents({
+    const timeout = new Timeout(10000)
+    showMessageWithCancel(`Loading log data from group ${logGroupInfo.groupName}`, timeout)
+    const responsePromise = client.filterLogEvents({
         logGroupName: logGroupInfo.groupName,
         filterPattern: parameters.filterPattern,
         nextToken,
         limit: parameters.limit,
     })
-
-    // Use heuristic of last token as backward token and next token as forward to generalize token form.
-    // Note that this may become inconsistent if the contents of the calls are changing as they are being made.
-    // However, this fail wouldn't really impact customers.
-    return {
-        events: response.events ? response.events : [],
-        nextForwardToken: response.nextToken,
-        nextBackwardToken: nextToken,
+    let response
+    try {
+        response = await waitTimeout(responsePromise, timeout)
+        // Use heuristic of last token as backward token and next token as forward to generalize token form.
+        // Note that this may become inconsistent if the contents of the calls are changing as they are being made.
+        // However, this fail wouldn't really impact customers.
+        if (response) {
+            return {
+                events: response.events ? response.events : [],
+                nextForwardToken: response.nextToken,
+                nextBackwardToken: nextToken,
+            }
+        }
+    } catch (err) {
+        if (!CancellationError.isUserCancelled(err)) {
+            getLogger().error(`cloudwatchlogs: failed to recieve log data for ${logGroupInfo}`)
+            throw err
+        } else {
+            getLogger().info('cloudwatchlogs: user cancelled search')
+        }
```

Comment:
Don't catch anything. We've made the signature of `filterLogEventsFromUriComponents` more complicated by doing this. Catch cancellation errors at your entry-point instead, or whenever you need to clean-up state.

Reply:
If I understand correctly, we want to only catch `CancellationError` here and not any other type of error? How do I implement that? I tried `catch(err: CancellationError)` but that doesn't seem to be correct. Also, what exactly does the signature of `filterLogEventsFromUriComponents` refer to?